homeassistant/components/livisi/switch.py

```diff
@@ -0,0 +1,185 @@
+        capability_id=capabilities[0],
+        is_on=is_on,
+        room=room_name,
+    )
+    return livisi_switch
+
+
+class LivisiSwitch(SwitchEntity, CoordinatorEntity):
+    """Represents the Livisi Switch."""
+
+    def __init__(
+        self,
+        hass,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        unique_id,
+        manufacturer,
+        product,
+        serial_number,
+        device_type,
+        name,
+        capability_id,
+        is_on,
+        room,
+        version=None,
+    ):
+        """Initialize the Livisi Switch."""
+        self.hass: HomeAssistant = hass
+        self.config_entry: ConfigEntry = config_entry
+        self._attr_unique_id = unique_id
+        self._manufacturer = manufacturer
+        self._product = product
+        self._serial_number = serial_number
+        self._attr_model = device_type
+        self._attr_name = name
+        self._state = None
+        self._capability_id = capability_id
+        self._attr_is_on = is_on
+        self._room = room
+        self._version = version
+        self.aio_livisi = coordinator.aiolivisi
+        if is_on is None:
+            self._attr_is_available = False
+        else:
+            self._attr_is_available = True
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(self._attr_unique_id))},
+            manufacturer=self._manufacturer,
+            model=self._attr_model,
+            name=self._attr_name,
+            suggested_area=self._room,
+            sw_version=self._version,
+            via_device=(DOMAIN, str(self._attr_unique_id)),
+        )
+        super().__init__(coordinator)
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return the device state."""
+        return self._attr_is_on
+
+    @property
+    def available(self) -> bool:
+        """Return if switch is available."""
+        return self._attr_is_available
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the entity on."""
+        if self.is_on is True:
+            return
+        response = await self.aio_livisi.async_pss_set_state(
+            self._capability_id, is_on=True
+        )
+        if response is None:
+            self._attr_is_available = False
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the entity off."""
+        if self.is_on is False:
+            return
+        response = await self.aio_livisi.async_pss_set_state(
+            self._capability_id, is_on=False
+        )
+        if response is None:
+            self._attr_is_available = False
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        self.coordinator.async_add_listener(self.update_states)
+        self.coordinator.async_add_listener(self.update_reachability)
```

Comment:
We need to unsubscribe listeners when removing the entity. We can use `self.async_on_remove`.

But this listener needs to receive the same state as the coordinator data. We can't set different state for different listeners in the same coordinator. Different listeners can extract different items from the state, but the passed state needs to be the same for all listeners of one coordinator.

Reply:
Now the state and reachability update is done by using the dispatcher. I don't think I need to use `self.async_on_remove` anymore. Am I right?